src/components/AllProducts.js

```diff
@@ -0,0 +1,66 @@
+import React, { useState } from "react";
+import Product from "./Product";
+import Sidebar from "./Sidebar";
+import Paging from "./Paging";
+
+const AllProducts = ({ products, categories }) => {
+  const [sidebarCategories, setItems] = useState(categories);
+  const [shownProducts, setProducts] = useState(products);
+
+  const updateFilter = (id) => {
+    let newSidebarCategories = [];
+    let filter = [];
+    for (var categoriesCounter in sidebarCategories) {
+      var category = sidebarCategories[categoriesCounter];
+
+      if (category.id === id) {
+        category.selected = !category.selected;
+      }
+
+      newSidebarCategories.push({
+        name: category.name,
+        id: category.id,
+        selected: category.selected,
+      });
+
+      if (category.selected) {
+        filter.push(category.id);
+      }
+    }
```

Comment:
Using `map` instead of `for` simplifies code a lot, along with [spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) and [ternary operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator)
```suggestion
    let filter = [];
    let newSidebarCategories = sidebarCategories.map((category) => {
      const newCategory = {
        ...category,
        selected: category.id === id ? !category.selected : category.selected,
      };

      if (newCategory.selected) 
        filter.push(category.id);
      return newCategory;
    });
    }
```